packages/blob/src/api.ts

```diff
@@ -252,10 +247,12 @@ async function getBlobError(
   return { code, error };
 }
 
+const CHUNK_SIZE = 64 * 1024;
+
 export async function requestApi<TResponse>(
   pathname: string,
   init: RequestInit,
-  commandOptions: BlobCommandOptions | undefined,
+  commandOptions: (BlobCommandOptions & WithUploadProgress) | undefined,
```

Comment:
I initially thought onUploadProgress should be added to BlobCommandOptions, but BlobCommandOptions is shared between public and private interfaces for all blob commands. While onUploadProgress is available only for put/upload. If we add it to BlobCommandOptions then it will show as an optional options for all commands (even head)